### ISSUE TITLE
[V3 Config] Config does not reject bad keys

### ISSUE THREAD

**Issue by mikeshardmind** [Cog-Creators/Red-DiscordBot#2038]
Including a `.` or starting a keyname with `$` needs to be invalid in config because of mongoDB backend support. 

Config will work just fine for this on the json backend, but not with the mongodb one because these characters are not allowed in field names for mongo. Instead, it works with json, and throws uncaught exceptions on mongo.

I'd be in favor of limiting keynames to alpha-numeric and space, underscore, and hyphens. to prevent future issues like this with other future possible backends without possibly breaking cogs twice for this.

**Comment by Tobotimus** [comment]
Seems liks `.` is fine, the only issue is if a key starts with `$`, see here: https://docs.mongodb.com/manual/reference/limits/#Restrictions-on-Field-Names.

I'm not comfortable with limiting config keys to such a small subset, there are so many places where this would warrant a redesign of config backends. Moreover it creates headaches when setting dicts as values. I'd prefer addressing this on a driver-to-driver basis and trying to escape special characters as required by each driver.

**Comment by mikeshardmind** [comment]
From your very link:
```
Until support is added in the query language, the use of $ and . in field names is not recommended and is not supported by the official MongoDB drivers.
```

side: I dislike per driver handling unless this is escaping, not rejecting as already stated.

**Comment by Tobotimus** [comment]
Resolved by #2212.

Style from now should be to try and escape special characters rather than reject them.
